### ISSUE TITLE
shard] 2016/10/05 15:06:17 Error collecting shard size: open /path to influx directory/dbname/telemetry/63: too many open files.

### ISSUE THREAD

**Issue by venkat10** [influxdata/influxdb#7417]
I'm having a problem with influxdb, due to the below error, my influxdb is hung and need to restart everytime.

Attached system details and configuration file, please find the attachments.
[server_info.docx](https://github.com/influxdata/influxdb/files/512707/server_info.docx)
[influxdb_config.docx](https://github.com/influxdata/influxdb/files/512708/influxdb_config.docx)

error:
[shard] 2016/10/05 15:06:17 Error collecting shard size: open /path to influx directory/dbname/telemetry/63: too many open files.

lsof|wc -l
14149

The following are open files limits:
ulimit -a
core file size (blocks, -c) 97
data seg size (kbytes, -d) unlimited
scheduling priority (-e) 0
file size (blocks, -f) unlimited
pending signals (-i) 31134
max locked memory (kbytes, -l) 64
max memory size (kbytes, -m) unlimited
open files (-n) 65000
pipe size (512 bytes, -p) 8
POSIX message queues (bytes, -q) 819200
real-time priority (-r) 0
stack size (kbytes, -s) 8192
cpu time (seconds, -t) unlimited
max user processes (-u) 4096
virtual memory (kbytes, -v) unlimited
file locks (-x) unlimited﻿

**Comment by beckettsean** [comment]
Originally from https://plus.google.com/u/0/105739473734911368561/posts/AusgZQa5p9a

Pulling data from the word files:

> 1) version installed, and from what package

version installed - 1.0
Package : https://dl.influxdata.com/influxdb/releases/influxdb-1.0.0_linux_amd64.tar.gz

> 2) Operating system and version

Redhat Linux, Version RHEL7.2

> 3) Fresh install or upgraded from prior versions?

upgraded from 0.13.0-1

> 4) The output of SHOW SHARDS

(attached as text file)

> 5) The output of SHOW RETENTION POLICIES ON (whatever the database name is):

```
> show databases
Name: databases
---------------
Name
cda_infra_db_prod
_internal
telegraf
emtlp
cda_app_db_dev
cda_app_db_prod
cda_app_db_stage

> SHOW RETENTION POLICIES on cda_infra_db_prod
name            duration        shardGroupDuration      replicaN        default
default         0               168h0m0s                1               false
telemetry       2160h0m0s       24h0m0s                 1               true

> SHOW RETENTION POLICIES on telegraf
name    duration        shardGroupDuration      replicaN        default
default 0               168h0m0s                1               true

> SHOW RETENTION POLICIES on emtlp
name    duration        shardGroupDuration      replicaN        default
default 0               168h0m0s                1               true

> SHOW RETENTION POLICIES on cda_app_db_dev
name            duration        shardGroupDuration      replicaN        default
default         0               168h0m0s                1               false
telemetry       2160h0m0s       24h0m0s                 1               true

> SHOW RETENTION POLICIES on cda_app_db_stage
name            duration        shardGroupDuration      replicaN        default
default         0               168h0m0s                1               false
telemetry       2160h0m0s       24h0m0s                 1               true

> SHOW RETENTION POLICIES on cda_app_db_prod
name            duration        shardGroupDuration      replicaN        default
default         0               168h0m0s                1               false
telemetry       2160h0m0s       24h0m0s                 1               true
```

> 6) `lsof | grep influxdb | wc -l`

11645

**Comment by beckettsean** [comment]
[show shards.txt](https://github.com/influxdata/influxdb/files/512801/show.shards.txt)

**Comment by beckettsean** [comment]
config file:

```
reporting-disabled = false
bind-address = ":8088"

[meta]
  dir = "/apps/dfttel/.influxdb/meta"
  retention-autocreate = true
  logging-enabled = true

[data]
  dir = "/apps/dfttel/.influxdb/data"
  engine = "tsm1"
  wal-dir = "/apps/dfttel/.influxdb/wal"
  wal-logging-enabled = true
  query-log-enabled = true
  cache-max-memory-size = 524288000
  cache-snapshot-memory-size = 26214400
  cache-snapshot-write-cold-duration = "1h0m0s"
  compact-full-write-cold-duration = "24h0m0s"
  max-points-per-block = 0
  max-series-per-database = 1000000
  trace-logging-enabled = false

[coordinator]
  write-timeout = "5s"
  max-concurrent-queries = 0
  query-timeout = "0"
  log-queries-after = "0"
  max-select-point = 0
  max-select-series = 0
  max-select-buckets = 0

[retention]
  enabled = true
  check-interval = "30m0s"

[shard-precreation]
  enabled = true
  check-interval = "10m0s"
  advance-period = "30m0s"

[admin]
  enabled = true
  bind-address = ":8083"
  https-enabled = false
  https-certificate = "/etc/ssl/influxdb.pem"

[monitor]
  store-enabled = true
  store-database = "_internal"
  store-interval = "10s"

[subscriber]
  enabled = true
  http-timeout = "30s"

[http]
  enabled = true
  bind-address = ":8086"
  auth-enabled = false
  log-enabled = true
  write-tracing = false
  https-enabled = false
  https-certificate = "/etc/ssl/influxdb.pem"
  https-private-key = ""
  max-row-limit = 10000
  max-connection-limit = 0
  shared-secret = ""
  realm = "InfluxDB"

[[graphite]]
  enabled = false

[[collectd]]
  enabled = false

[[opentsdb]]
  enabled = false

[[udp]]
  enabled = false

[continuous_queries]
  log-enabled = true
  enabled = true
  run-interval = "1s"
```

**Comment by beckettsean** [comment]
@venkat10 I would recommend using longer than 1 day shards with 90 day RPs. The fewer shards the better, generally speaking. More shards means more RAM use, more open files, more directories, and more index duplication.

**Comment by jwilder** [comment]
#7335 likely fixed this an is 1.0.2.  Please upgrade and if this issue is still occurring comment on this issue and we'll re-open.  Increasing the shard duration from 1 day as @beckettsean suggests will also be very worthwhile.
